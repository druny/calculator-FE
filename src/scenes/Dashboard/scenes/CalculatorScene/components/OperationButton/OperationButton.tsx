import React, { Dispatch } from 'react';
import { SvgIcon, Button } from '@material-ui/core';
import { SvgIconComponent } from '@material-ui/icons';

type OperationButtonPropsType = {
  onClick: Dispatch<string>;
  operationName: string;
  operationLogo: SvgIconComponent;
};

export function OperationButton({
  onClick,
  operationName,
  operationLogo,
}: OperationButtonPropsType) {
  return (
    <Button variant="outlined" color="secondary" onClick={() => onClick(operationName)}>
      <SvgIcon component={operationLogo} />
    </Button>
  );
}
