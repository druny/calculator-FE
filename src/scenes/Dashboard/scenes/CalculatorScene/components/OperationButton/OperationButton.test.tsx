import React from 'react';
import renderer from 'react-test-renderer';
import { SvgIcon } from '@material-ui/core';

import { OperationButton } from './OperationButton';

it('renders correctly when there are no items', () => {
  const tree = renderer
    .create(<OperationButton operationLogo={SvgIcon} onClick={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
