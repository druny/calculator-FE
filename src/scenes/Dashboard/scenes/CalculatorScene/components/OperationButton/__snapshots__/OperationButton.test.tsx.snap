// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    />
  </span>
</button>
`;
